design/bridge_design.md

```diff
@@ -0,0 +1,317 @@
+
+```solidity
+function publishMessage(
+        uint32 nonce, //deduplication/uniqueness mechanism
+        bytes memory topic,
+        bytes memory payload,
+        uint8 consistencyLevel //number of confirmation blocks to wait for
+) public payable returns (uint64 sequence) {
+    // check fee
+    require(msg.value == messageFee(), "invalid fee");
+
+    sequence = useSequence(msg.sender);
+    // emit log
+    emit LogMessagePublished(msg.sender, sequence, nonce, topic, payload);
+}
+```
+
+Any contract or user can call the `publishMessage` function. Any message passed will be bound to its sender so contracts cannot simply impersonate one another. As messages are not stored, but rather emitted as events all of our synchronization behind the scenes will happen inside of the `Enclave`. It will be subscribed to those events. 
+
+`nonce` is a deduplication mechanism that enables uniqueness between two identical payloads on the same topic. An example is when the message __"I've received 50 USDC tokens from 0xBob"__ is received twice with different nonces, this should mean those were two different deposits. 
+
+`consistencyLevel` is a mechanism to introduce delay before the message is considered valid. The platform will only expose the message after the block producing it is confirmed by blocks equal to the `consistencyLevel`
+
+> **_NOTE:_**  `consistencyLevel` is only required if Obscuro does not reorganize when L1 deposits reorganize. In other instances it can be set to 0 or 1. There is however another use case for it described in [Security](#MessageBusSecurity).
+
+
+### Verify Message
+
+This function checks if such a message has been submitted from the other network to the `MessageBus` contract and is useful for pull semantics. Its interface will look like this:
+
+```solidity 
+function verifyMessageReceived(
+        address      sender,
+        uint??       sequence,
+        uint32       nonce,
+        bytes memory topic,
+        bytes memory payload
+) public returns (bool)
+```
+
+Internally, the function will hash the message and compare it with the result of the key in `receivedMessages` map. If the map contains a `true` under this key, then the message has been received by this contract and `verifyMessageReceived` will return true. This is useful as a cheap-ish way to verify something has happened by utilizing the end user's client (or browser) as the transfer mechanism for the full message.  
+
+
+
+### Submit Out Of Network Messages
+
+This is the administrative smart contract function that will be called by the `ManagementContract` on L1 and the `enclave` on L2. It's purpose is to store the messages that were submitted from the other linked chain. It should be access controlled and called according to the `consistencyLevel` and Obscuro platform rules.
```

Comment:
It's a bit misleading to call this "administrative"
I think a better introduction is to say that this is the function used to push messages into the MessageBus.